src/core/parser/component.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  transformStylesAssignmentsToJson,
+  transformAttributesAssignmentsToJson,
+  transformStringAssignmentToJson,
+} from '../../shared/utils/uidl-utils'
+
+import { ComponentUIDL, UIDLNode } from '../../typings/uidl-definitions'
+
+export const parseComponentJSON = (input: Record<string, unknown>): ComponentUIDL => {
+  const node = input.node as Record<string, unknown>
+  const result: ComponentUIDL = {
+    ...(input as ComponentUIDL),
+  }
+
+  switch (node.type) {
+    case 'element':
+      const elementContent = node.content as Record<string, unknown>
+
+      if (elementContent.style) {
+        elementContent.style = transformStylesAssignmentsToJson(elementContent.style as Record<
+          string,
+          unknown
+        >)
+      }
+
+      if (elementContent.attrs) {
+        elementContent.attrs = transformAttributesAssignmentsToJson(elementContent.attrs as Record<
+          string,
+          unknown
+        >)
+      }
+
+      if (Array.isArray(elementContent.children)) {
+        elementContent.children = elementContent.children.reduce(
+          (newChildren: UIDLNode[], child) => {
+            if (typeof child === 'string') {
+              newChildren.push(transformStringAssignmentToJson(child))
+            } else {
+              newChildren.push(parseComponentJSON({ node: child }).node)
```

Comment:
why not parseComponentJSON(child) ?

Reply:
oh good point, this evolved into the current api and I didn;t get to change it.

Reply:
actually because `parseComponentJSON({ node: child })` returns a component without anything but a node key. parseComponentJSON(child) is invalid because child is a `UIDLnode`, not a `UIDLComponent `

Reply:
so it will remain {node: child} ?

I was asking about `{node: child}.node` which is equivalent with `child`